connectors/src/lib/models/webcrawler.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  type CreationOptional,
+  DataTypes,
+  type ForeignKey,
+  type InferAttributes,
+  type InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { Connector, sequelize_conn } from "@connectors/lib/models";
+
+export class WebCrawlerConfiguration extends Model<
+  InferAttributes<WebCrawlerConfiguration>,
+  InferCreationAttributes<WebCrawlerConfiguration>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare url: string;
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+WebCrawlerConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    url: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    indexes: [],
+    modelName: "webcrawler_configurations",
+  }
+);
+Connector.hasMany(WebCrawlerConfiguration);
+
+export class WebCrawlerFolder extends Model<
+  InferAttributes<WebCrawlerFolder>,
+  InferCreationAttributes<WebCrawlerFolder>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare parentUrl: string | null;
+  declare url: string;
+  // Folders are not upserted to the data source but their ids are
+  // used as parent to WebCrawlerPage.
+  declare dustDocumentId: string;
+  declare connectorId: ForeignKey<Connector["id"]>;
+  declare webcrawlerConfigurationId: ForeignKey<WebCrawlerConfiguration["id"]>;
+}
+
+WebCrawlerFolder.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    url: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    parentUrl: {
```

Comment:
I think it'll make our life easier to simply store parentInternalId ?

Reply:
not necessarily, the real id is the URL itself, so it makes sense to store the URL.
The hash is just a way to make it more compatible with other API (aka the Dust API).